### PR TITLE
[WIP] Tf2estimator on Pyspark backend big weights support

### PR DESCRIPTION
**Pull request by jenniew**
Tf2estimator on Pyspark backend big weight support

### CODE REVIEW

File: python/orca/src/bigdl/orca/learn/tf2/spark_runner.py [hunk 1]
```diff
@@ -344,9 +347,11 @@ def step(self, data_creator, epochs=1, batch_size=32, verbose=1,
         else:
             stats = {k: v[-1] for k, v in history.history.items()}
         if self.rank == 0:
-            # if model_dir is not None:
-            #     model.save_weights(model_dir)
-            return ([weights, stats])
+            if self.model_dir is not None:
```
Comment (yangw1234) [inline python/orca/src/bigdl/orca/learn/tf2/spark_runner.py:350]: how about just using `torch.save(model.state_dict(), PATH)`

Comment (jenniew) [reply python/orca/src/bigdl/orca/learn/tf2/spark_runner.py:350]: torch.save can support hdfs and s3?

Comment (yangw1234) [reply python/orca/src/bigdl/orca/learn/tf2/spark_runner.py:350]: I meaning inside `save_pkl` and `load_pkl`?

File: python/orca/src/bigdl/orca/learn/tf2/pyspark_estimator.py [hunk 2]
```diff
@@ -151,34 +152,36 @@ def transform_func(iter, init_param, param):
                     param["validation_data_creator"] = make_data_creator(valid_list)
                     return SparkRunner(**init_param).step(**param)
 
-                res = data.zip(validation_data).rdd.repartition(self.num_workers).barrier() \
+                self.states_rdd = data.zip(validation_data).rdd.repartition(self.num_workers).barrier() \
```
Comment (yangw1234) [inline python/orca/src/bigdl/orca/learn/tf2/pyspark_estimator.py]: I think there is an assumption of this line that `data` and `validataion_data` has the same number of partitions and the same number of elements on each partition.
I think the assumption may not hold if `OrcaContext._shard_size` is not None.

File: python/orca/src/bigdl/orca/learn/tf2/spark_runner.py (hunk 1 above)
Comment (yangw1234) [reply python/orca/src/bigdl/orca/learn/tf2/spark_runner.py:356]: Sorry. This is TF. Orz

**Comment by jenniew** [comment]
Jenkins pass: http://10.112.231.51:18888/view/ZOO-PR/job/BigDL-Orca-PR-Validation/92/

**Pull request merged**